server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -0,0 +1,194 @@
+    public void clusterChanged(ClusterChangedEvent event) {
+        // wait for the cluster state to be recovered
+        if (event.state().blocks().hasGlobalBlock(GatewayService.STATE_NOT_RECOVERED_BLOCK)) {
+            return;
+        }
+
+        DiscoveryNode healthNode = HealthNode.findHealthNode(event.state());
+        if (healthNode == null) {
+            return;
+        }
+        final boolean isCurrentlyHealthNode = healthNode.getId().equals(this.clusterService.localNode().getId());
+
+        final boolean prevIsHealthNode = this.isHealthNode;
+        if (prevIsHealthNode != isCurrentlyHealthNode) {
+            this.isHealthNode = isCurrentlyHealthNode;
+            if (this.isHealthNode) {
+                // we weren't the health node, and now we are
+                maybeScheduleJob();
+            } else {
+                // we were the health node, and now we aren't
+                cancelJob();
+            }
+        }
+    }
+
+    @Override
+    public void close() {
+        SchedulerEngine engine = scheduler.get();
+        if (engine != null) {
+            engine.stop();
+        }
+    }
+
+    private void cancelJob() {
+        if (scheduler.get() != null) {
+            scheduler.get().remove(HEALTH_SERVICE_JOB_NAME);
+            scheduledJob = null;
+        }
+    }
+
+    private boolean isClusterServiceStoppedOrClosed() {
+        final Lifecycle.State state = clusterService.lifecycleState();
+        return state == Lifecycle.State.STOPPED || state == Lifecycle.State.CLOSED;
+    }
+
+    private void maybeScheduleJob() {
+        if (this.isHealthNode == false) {
+            return;
+        }
+
+        // don't schedule the job if the node is shutting down
+        if (isClusterServiceStoppedOrClosed()) {
+            logger.trace(
+                "Skipping scheduling a HealthService job due to the cluster lifecycle state being: [{}] ",
+                clusterService.lifecycleState()
+            );
+            return;
+        }
+
+        if (scheduler.get() == null) {
+            scheduler.set(new SchedulerEngine(settings, clock));
+            scheduler.get().register(this);
+        }
+
+        assert scheduler.get() != null : "scheduler should be available";
+        scheduledJob = new SchedulerEngine.Job(HEALTH_SERVICE_JOB_NAME, new TimeValueSchedule(pollInterval));
+        scheduler.get().add(scheduledJob);
+    }
+
+    @Override
+    public void triggered(SchedulerEngine.Event event) {
+        if (event.getJobName().equals(HEALTH_SERVICE_JOB_NAME)) {
+            if (this.isHealthNode) {
+
+                this.healthService.getHealth(this.client, null, true, 0, new ActionListener<List<HealthIndicatorResult>>() {
+                    @Override
+                    public void onResponse(List<HealthIndicatorResult> healthIndicatorResults) {
+                        GetHealthAction.Response healthResponse = new GetHealthAction.Response(null, healthIndicatorResults, true);
```

Comment:
It's strange here to construct a response object that's not used. Perhaps we should use `healthIndicatorResults.forEach(…)` below, and then explicitly call `HealthStatus.merge(…)` to get the top-level status?